src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/CompiledPageRouteModelProvider.cs

```diff
@@ -58,26 +69,63 @@ private void EnsureCache()
                     rootDirectory = rootDirectory + "/";
                 }
 
+                var areaRootDirectory = _pagesOptions.AreaRootDirectory;
```

Comment:
Consider this scenario:

```
/Areas/First.cshtml
/Areas/MyArea/Pages/Second.cshtml
```

where `RootDirectory = /, AreaRootDirectory = /Areas`

Should we

1) Disallow overlapping of `RootDirectory` and `AreaRootDirectory`
2) Disallow `RootDirectory` based routes for things that match an area path. In this case, we'll end up with `/Areas/First` and `/MyArea/Second` as the routes
3) Don't do anything special for area paths. In this case, we'll end up with `/Areas/First`, `/Areas/MyArea/Pages/Second`, and `/MyArea/Second` as the routes

IMO, option 2, seems like the most appropriate behavior.

Reply:
If I understand this correctly, you want to prevent people from havign a page like /Areas/First.cshtml? If that's the case, I'm ok with it. That said, I think I'm also fine not doing anything as you need to go out of your way to shoot yourself in the foot.

Reply:
I spoke with @DamianEdwards and @Eilon  about this today. We agreed with 2.

Reply:
Tracking via https://github.com/aspnet/Mvc/issues/7147